client/v3/balancer/balancer.go

```diff
@@ -191,6 +216,24 @@ func (bb *baseBalancer) HandleResolvedAddrs(addrs []resolver.Address, err error)
 			// (DO NOT) delete(bb.scToSt, sc)
 		}
 	}
+
+	// TODO: Consider just returning `ErrBadResolverState` if `warnedErrors` is
+	//       not empty. Or (to make a breaking change) just early exit with a
+	//       `ErrBadResolverState` when `warnedErrors` would first be appended to.
```

Comment:
@ptabor Is this the what/where you had in mind for a TODO?

Reply:
I would remove this: 
> Or (to make a breaking change) just early exit with a ...

I assume we want to make the connection to etcd as long as we can connect to at least 1 endpoint, 
so we should never exit early.

Reply:
Ah sorry I missed this comment!

Reply:
OK just rebased / squashed without the "Or ..." part of this `TODO`.